lib/esbonio/esbonio/lsp/rst/__init__.py

```diff
@@ -193,6 +194,18 @@ def __repr__(self):
         )
 
 
+class HoverContext:
+    """A class that captures the context within a hover request has been made."""
+
+    def __init__(self, *, doc: Document, position: Position):
```

Comment:
Can this be extended to take an instance of `ClientCapabilities` so language features can check what the client supports (much like how `CompletionContext` does things)